docker/README.md

```diff
@@ -208,4 +211,57 @@ healthcheck:
 ## Using jemalloc
 
 Jemalloc is embedded in the image and will be used instead of the default allocator.
-You can read about jemalloc by reading the Synapse [README](../README.md)
\ No newline at end of file
+You can read about jemalloc by reading the Synapse [README](../README.md)
+
+### Running all worker processes in a single container for testing
+
+Anyone looking to run Synapse workers with docker in production should use one container
+per Synapse process, specifying a different worker config per container.
+Running all Synapse processes is possible, but is currently only used for testing 
+purposes. [Dockerfile-workers](Dockerfile-workers) will produce an image bundling
+all necessary components together for a workerised homeserver instance.
+
+This includes any desired Synapse worker processes, a nginx to route traffic accordingly,
+a redis for worker communication and a supervisord instance to start up and monitor all
+processes. You will need to provide your own postgres container to connect to. TLS is not
+handled by the container. A external reverse proxy should be placed in front the exposed
+HTTP port for the purposes of TLS termination.
+
+Note that the worker Dockerfile is based off the main `[Dockerfile](Dockerfile)`. If you
+would like to build from the current checkout, first build the main Synapse docker image
+using the instructions in [Building the image](#building-the-image), then afterwards build
+the worker image:
+
+```
+docker build -t matrixdotorg/synapse:workers -f docker/Dockerfile-workers .
```

Comment:
given it's built from a different Dockerfile, I think it should have a different image name, rather than just being a tag on the regular image:

```suggestion
docker build -t matrixdotorg/synapse-workers -f docker/Dockerfile-workers .
```